tests/MongoDB.Bson.Tests/ObjectModel/BsonBinaryDataTests.cs

```diff
@@ -15,13 +15,108 @@
 
 using System;
 using System.Linq;
-using MongoDB.Bson;
+using FluentAssertions;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
 using Xunit;
 
 namespace MongoDB.Bson.Tests
 {
     public class BsonBinaryDataTests
     {
+        [Theory]
+        [ParameterAttributeData]
+        public void constructor_should_throw_when_bytes_is_null(
+            [Range(1, 3)] int overload)
+        {
+            var bytes = (byte[])null;
+
+            Exception exception = null;
+            switch (overload)
+            {
+                case 1: exception = Record.Exception(() => new BsonBinaryData(bytes)); break;
+                case 2: exception = Record.Exception(() => new BsonBinaryData(bytes, BsonBinarySubType.Binary)); break;
+                case 3: exception = Record.Exception(() => new BsonBinaryData(bytes, BsonBinarySubType.Binary, GuidRepresentation.Unspecified)); break;
+            }
```

Comment:
I haven't used this particular strategy before... the scenario is that there are 3 overloads that need to be tested with the same Arrange and Assert code, so this is a way to use one test method to test multiple overload.